--- loudmouth/loudmouth/lm-connection.c.orig	2003-10-13 10:28:59.000000000 +0000
+++ loudmouth/loudmouth/lm-connection.c	2003-11-22 14:27:10.000000000 +0000
@@ -345,7 +345,7 @@
 					connection->gnutls_xcred);
 		
 		gnutls_transport_set_ptr (connection->gnutls_session, 
-					  (gnutls_transport_ptr) fd);
+					  (gnutls_transport_ptr) (size_t) fd);
 
 		ret = gnutls_handshake (connection->gnutls_session);
 		
@@ -426,7 +426,7 @@
 		     LmConnection *connection)
 {
 	gchar     buf[IN_BUFFER_SIZE];
-	gint      bytes_read;
+	gsize     bytes_read;
 	GIOStatus status;
        
 	if (!connection->io_channel) {
@@ -491,7 +491,7 @@
 	g_log (LM_LOG_DOMAIN, LM_LOG_LEVEL_NET, 
 	       "-----------------------------------\n");
 
-	lm_verbose ("Read: %d chars\n", bytes_read);
+	lm_verbose ("Read: %d chars\n", (int)bytes_read);
 
 	lm_parser_parse (connection->parser, buf);
 	
