diff -urN aa/loudmouth-0.15.1/loudmouth/lm-ssl.c loudmouth-0.15.1/loudmouth/lm-ssl.c
--- aa/loudmouth-0.15.1/loudmouth/lm-ssl.c	2004-01-23 05:18:27.000000000 +0100
+++ loudmouth-0.15.1/loudmouth/lm-ssl.c	2004-02-22 16:12:21.819050016 +0100
@@ -71,7 +71,6 @@
 	}
 	
 	if (status & GNUTLS_CERT_INVALID
-	    || status & GNUTLS_CERT_NOT_TRUSTED
 	    || status & GNUTLS_CERT_REVOKED) {
 		if (ssl->func (ssl, LM_SSL_STATUS_UNTRUSTED_CERT,
 			       ssl->func_data) != LM_SSL_RESPONSE_CONTINUE) {
