diff -urN loudmouth-1.4.3/configure.ac loudmouth-1.4.3.new/configure.ac
--- loudmouth-1.4.3/configure.ac	2008-10-29 21:23:52.000000000 +0100
+++ loudmouth-1.4.3.new/configure.ac	2009-10-24 14:59:47.623887421 +0200
@@ -146,7 +146,7 @@
 enable_ssl=no
 if test "x$ac_ssl" = "xgnutls"; then
   dnl Look for GnuTLS
-  AM_PATH_LM_LIBGNUTLS($GNUTLS_REQUIRED, have_libgnutls=yes, have_libgnutls=no)
+  PKG_CHECK_MODULES(LIBGNUTLS, gnutls >= $GNUTLS_REQUIRED, have_libgnutls=yes, have_libgnutls=no)
   if test "x$have_libgnutls" = "xyes"; then
     CFLAGS="$CFLAGS $LIBGNUTLS_CFLAGS"
     LIBS="$LIBS $LIBGNUTLS_LIBS"
