--- loudmouth-1.4.3/configure.ac.orig	2013-11-09 15:27:13.353830101 +0100
+++ loudmouth-1.4.3/configure.ac	2013-11-09 15:36:11.977140831 +0100
@@ -120,9 +120,9 @@
 dnl +--------------------------------------------------------+
 dnl | Checking for libnsl and libsocket need on some systems |-
 dnl +--------------------------------------------------------+
-AC_CHECK_LIB(nsl,gethostbyname)
-AC_CHECK_LIB(socket,socket)
-AC_CHECK_LIB(resolv,res_query)
+AC_SEARCH_LIBS(gethostbyname,nsl)
+AC_SEARCH_LIBS(socket,socket)
+AC_SEARCH_LIBS(res_query,resolv)
 
 dnl +--------------------------------------------------------+
 dnl | Checking for SSL support                               |-
