diff -Nur loudmouth-0.15.1.orig/loudmouth/lm-connection.c loudmouth-0.15.1/loudmouth/lm-connection.c
--- loudmouth-0.15.1.orig/loudmouth/lm-connection.c	2004-01-21 16:53:08.000000000 +0100
+++ loudmouth-0.15.1/loudmouth/lm-connection.c	2004-01-23 12:37:39.336766584 +0100
@@ -595,7 +595,7 @@
 		     LmConnection *connection)
 {
 	gchar     buf[IN_BUFFER_SIZE];
-	gint      bytes_read;
+	gsize     bytes_read;
 	GIOStatus status;
        
 	if (!connection->io_channel) {
@@ -644,7 +644,7 @@
 	g_log (LM_LOG_DOMAIN, LM_LOG_LEVEL_NET, 
 	       "-----------------------------------\n");
 
-	lm_verbose ("Read: %d chars\n", bytes_read);
+	lm_verbose ("Read: %d chars\n", (int)bytes_read);
 
 	lm_parser_parse (connection->parser, buf);
 	
